src/utils/pathConflict.ts

```diff
@@ -0,0 +1,116 @@
+import fs from 'fs';
+import path from 'path';
+
+import semver from 'semver';
+import * as childProcess from '@/utils/childProcess';
+import resources from '@/resources';
+
+const flags: Record<string, string> = {
+  helm:    'version',
+  kim:     '-v',
+  kubectl: 'version',
+};
+const regexes: Record<string, RegExp> = {
+  // helm has to match both
+  // current: version.BuildInfo{Version:"v3.5.3", ...
+  // older:   Client: &version.Version{SemVer:"v2.16.12", ...
+  helm:    /Version.*:.*?"v(.+?)"/,
+  kim:     /version v(\S+)/,
+  kubectl: /Client Version.*?GitVersion:"v(.+?)"/,
+};
+
+export default async function pathConflict(targetDir: string, binaryName: string): Promise<Array<string>> {
+  const referencePath = resources.executable(binaryName);
+
+  try {
+    await fs.promises.access(referencePath, fs.constants.R_OK | fs.constants.X_OK);
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      console.log(err);
+
+      return [];
+    }
+  }
+  const proposedVersion = await getVersion(referencePath, binaryName);
+
+  if (!proposedVersion) {
+    return [];
+  }
+  const notes: Array<string> = [];
+  const paths: Array<string> = process.env.PATH?.split(path.delimiter) ?? [];
+
+  let sawCurrentDir = false;
+
+  for (const currentDir of paths) {
+    // canonicalize path names to avoid trailing slashes and '/./' sequences
+    // This is because users set the PATH environment variable, so
+    // we need to accommodate any irregularities.
+    // path.normalize doesn't remove a trailing slash, path.resolve does.
+    if (path.resolve(currentDir) === path.resolve(targetDir)) {
```

Comment:
Since `targetDir` is constant here, it might be useful to call `path.resolve()` on it before the loop. (But that's just an optimization; we don't really need that to merge the PR.)

Reply:
done